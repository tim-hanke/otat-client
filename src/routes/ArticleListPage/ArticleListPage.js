import React, { Component } from "react";
import ArticleListContext from "../../contexts/ArticleListContext";
import ArticleApiService from "../../services/article-api-service";
import { Section } from "../../components/Utils/Utils";
import ArticleListItem from "../../components/ArticleListItem/ArticleListItem";
import "./ArticleListPage.css";

export default class ArticleListPage extends Component {
  static contextType = ArticleListContext;

  componentDidMount() {
    this.context.clearError();
    ArticleApiService.getArticles()
      .then(this.context.setArticleList)
      .catch(this.context.setError);
  }

  renderArticles() {
    const { articleList = [] } = this.context;
    return articleList
      .sort((a, b) => (a.id < b.id ? 1 : -1))
      .map((article) => <ArticleListItem key={article.id} article={article} />);
  }

  render() {
    const { error } = this.context;
    return (
      <Section list className="ArticleListPage">
        {error ? (
          <p className="red">There was an error, try again</p>
        ) : (
          this.renderArticles()
        )}
      </Section>
    );
  }
}
