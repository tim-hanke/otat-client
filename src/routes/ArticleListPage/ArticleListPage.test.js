import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import ArticleListPage from "./ArticleListPage";

describe("ArticleListPage component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <ArticleListPage />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ArticleListPage />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
