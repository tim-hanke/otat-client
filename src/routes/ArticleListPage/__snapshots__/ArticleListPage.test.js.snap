// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListPage component renders the UI as expected 1`] = `
<section
  className="Section Section--list ArticleListPage"
/>
`;
